### ISSUE TITLE
Support installing WinGet on Docker for Windows containers 

### ISSUE THREAD

**Issue by JensNordenbro** [microsoft/winget-cli#4144]
### Description of the new feature / enhancement

Appx packages are not possible to install on Windows containers .

WinGet is an excellent means to acquire tools for containers. 

### Proposed technical implementation details

Provide a Windows Containers friendly package. xcopy - deployment zip for example.

**Comment by mdanish-kh** [comment]
Related discussions:
- https://github.com/microsoft/winget-cli/discussions/1961
- https://github.com/microsoft/winget-cli/discussions/1962

**Comment by JensNordenbro** [comment]
In a way the request is perhaps more about being able to install msix, winget is just what is in front of me for the moment

**Comment by benh57** [comment]
Would love to see this moved forward especially with server 2025 now including winget.

**Comment by lygstate** [comment]
Is this a duplicated of https://github.com/microsoft/winget-cli/issues/3037?

**Comment by JensNordenbro** [comment]
It is.

Fix it * 2!

**Comment by benh57** [comment]
The powershell process in #3037  is considered the official position by the project? That is ridiculous to close this for that. There should be a straightforward way to install winget in docker.

**Comment by JensNordenbro** [comment]
I just closed it to only have one copy open to make life easier to the devs (If there are any).

I can reoopen of you want me to but please consider my logic first

**Comment by gerneio** [comment]
@JensNordenbro the other issue was closed a year ago.

**Comment by JensNordenbro** [comment]
Ohh, now I see your point.

**Comment by jankap** [comment]
I can install winget inside a container of `mcr.microsoft.com/windows/server:ltsc2025` via (thanks to https://github.com/microsoft/winget-cli/issues/3037#issuecomment-2155167046)

```
# Fail fast & no progress bars
    $ErrorActionPreference = 'Stop'
    $ProgressPreference = 'SilentlyContinue'

    # https://github.com/microsoft/winget-cli/issues/3037
    New-Item -Force -ItemType 'directory' -Name 'winget-cli' -Path 'C:/temp'
    Invoke-WebRequest 'https://aka.ms/vs/16/release/vc_redist.x64.exe' -OutFile 'C:/temp/vc_redist.x64.exe' -UseBasicParsing -Proxy $env:http_proxy
    powershell -c C:/temp/vc_redist.x64.exe /install /quiet /norestart
    Invoke-WebRequest 'https://aka.ms/getwinget' -OutFile 'C:/temp/winget-cli/winget.zip' -UseBasicParsing -Proxy $env:http_proxy
    echo "downloaded winget"
    Expand-Archive -LiteralPath 'C:/temp/winget-cli/winget.zip' -DestinationPath 'C:/temp/winget-cli' -Force
    echo "expanded archive"
    Move-Item -Path 'C:/temp/winget-cli/AppInstaller_x64.msix' -Destination 'C:/temp/winget-cli/AppInstaller_x64.zip'
    Expand-Archive -LiteralPath 'C:/temp/winget-cli/AppInstaller_x64.zip' -DestinationPath 'C:/temp/winget-cli' -Force
    New-Item -Force -ItemType 'directory' -Name 'winget-cli' -Path 'C:/'
    Move-Item -Path 'C:/temp/winget-cli/winget.exe' -Destination 'C:/winget-cli/winget.exe'
    Move-Item -Path 'C:/temp/winget-cli/WindowsPackageManager.dll' -Destination 'C:/winget-cli/WindowsPackageManager.dll'
    Move-Item -Path 'C:/temp/winget-cli/resources.pri' -Destination 'C:/winget-cli'

    echo "Adding winget to PATH (current session + machine)"
    $wingetDir = 'C:\winget-cli'
    # Persist:
    try {
      $machPath = [Environment]::GetEnvironmentVariable('Path','Machine')
      if ($machPath -notmatch [regex]::Escape($wingetDir)) {
        [Environment]::SetEnvironmentVariable('Path', "$machPath;$wingetDir", 'Machine')
      }
    } catch { }

    # Immediate use in this job:
    if ($env:Path -notmatch [regex]::Escape($wingetDir)) {
      $env:Path += ";$wingetDir"
    }

    echo "Install complete"

    # Verify
    winget --version
```

You might have to adjust `$env:http_proxy`. 

and a version is shown:

```
PS C:\> winget --version
v1.11.510
```

But I can't install anything:

```
winget settings --enable ProxyCommandLineOptions

winget source update --proxy $env:http_proxy
Updating all sources...
Updating source: msstore...
Done
Updating source: winget...
Cancelled
```

I haven't cancelled it - happend automatically. 

```
PS C:\> winget source list --proxy $env:http_proxy
Name    Argument                                      Explicit
--------------------------------------------------------------
msstore https://storeedgefd.dsx.mp.microsoft.com/v9.0 false
winget  https://cdn.winget.microsoft.com/cache        false
```

My proxy is *not* blocking the winget cache, because `iwr https://cdn.winget.microsoft.com/cache/source.msix -Proxy $env:http_proxy -UseBasicParsing -OutFile source.msix` works...

I can even add the package, at least there's no error, but it doesn't change anything.
```
PS C:\> Add-AppxPackage -Path .\source.msix
PS C:\>
```

Interestingly, msstore stuff seems to work, at least a search does work, whereas winget does not
```
PS C:\> winget search git --proxy $env:http_proxy --verbose --source winget
Failed when opening source(s); try the 'source reset' command if the problem persists.
An unexpected error occurred while executing the command:
0x8a15000f : Data required by the source is missing

PS C:\> winget search git --proxy $env:http_proxy --verbose --source msstore
Name      Id           Version
------------------------------
My Git    9NLVK2SL2SSP Unknown
GitFiend  9NMNKLTSZNKC Unknown
GitLooker 9PK6TGX9T87P Unknown
```

Of couese, I've reset the sources, but no change:

```
PS C:\> winget source reset --force --proxy $env:http_proxy
Resetting all sources...Done
```

So long story short, is winget unusable on Windows Containers?

**Comment by manupillai308** [comment]
To anyone who is facing the same issue, I was able to get it working:
```dockerfile
FROM mcr.microsoft.com/windows/server:ltsc2025

# Install Visual C++ Redistributable (required by many native packages)
WORKDIR C:\\Downloads
ADD https://aka.ms/vs/16/release/vc_redist.x64.exe C:\\Downloads\\vcredist_x64.exe
RUN C:\\Downloads\\vcredist_x64.exe /install /passive /norestart /log out.txt

# Install winget from GitHub release
ADD https://github.com/microsoft/winget-cli/releases/download/v1.29.250/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle C:\\Downloads\\winget.zip
RUN powershell -Command \
    "Expand-Archive -Path C:\\Downloads\\winget.zip -DestinationPath C:\\Downloads\\winget_outer -Force; \
     Copy-Item C:\\Downloads\\winget_outer\\AppInstaller_x64.msix C:\\Downloads\\winget_x64.zip; \
     Expand-Archive -Path C:\\Downloads\\winget_x64.zip -DestinationPath C:\\winget -Force"
RUN setx /M PATH "%PATH%;C:\winget"

CMD ["powershell"]

```
This works well for my usecase where I wanted to install a bunch of dependencies. 
My host version: `Microsoft Windows [Version 10.0.26200.8457]`
